third_party/terraform/utils/bootstrap_utils_test.go

```diff
@@ -322,10 +322,9 @@ func BootstrapConfig(t *testing.T) *Config {
 	}
 
 	config := &Config{
-		Credentials: getTestCredsFromEnv(),
```

Comment:
Why are we removing Credentials here?

Reply:
because the bootstrap test doesn't work with User ADCs.

```
REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google-beta   master  8⬇  2✎  1+  2  $    make testacc TEST=./google-beta TESTARGS="-run TestAccDataSourceGoogleServiceAccountIdToken_impersonation"
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run TestAccDataSourceGoogleServiceAccountIdToken_impersonation -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceGoogleServiceAccountIdToken_impersonation
=== PAUSE TestAccDataSourceGoogleServiceAccountIdToken_impersonation
=== CONT  TestAccDataSourceGoogleServiceAccountIdToken_impersonation
    TestAccDataSourceGoogleServiceAccountIdToken_impersonation: bootstrap_utils_test.go:334: Bootstrapping failed. Unable to load test config: unable to parse credentials from 'true': json: cannot unmarshal bool into Go value of type google.credentialsFile
--- FAIL: TestAccDataSourceGoogleServiceAccountIdToken_impersonation (0.00s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta 3.380s
FAIL
make: *** [testacc] Error 1
 REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google-beta   master  8⬇  3✎  1+  2  $    env | grep GOOGLE
GOOGLE_REGION=us-central1
GOOGLE_BILLING_ACCOUNT=REDACTED
GOOGLE_ORG=REDACTED
GOOGLE_ZONE=us-central1-a
GOOGLE_PROJECT=REDACTED
GOOGLE_USE_DEFAULT_CREDENTIALS=true
GOOGLE_SERVICE_ACCOUNT=terraform@REDACTED.iam.gserviceaccount.com
```

Reply:
We'll have to either modify the bootstrap test or our testing environment, because removing this breaks our KMS datasource tests. Any ideas for the bootstrap test on your end?

Reply:
I think TeamCity is configured to specify the `credentials` field of the provider via the environment variables instead of using the default credentials which work nicely everywhere.

```
REDACTED  MCW0CDP3YY  ~  go  …  github.com  hashicorp  terraform-provider-google-beta   master  13⬇  3✎  1+  2  $    make testacc TEST=./google-beta TESTARGS="-run TestAccDataSourceGoogleKmsCryptoKey_basic"
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run TestAccDataSourceGoogleKmsCryptoKey_basic -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceGoogleKmsCryptoKey_basic
2020/10/06 23:48:50 [INFO] Authenticating using DefaultClient...
2020/10/06 23:48:50 [INFO]   -- Scopes: [https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/cloud-identity https://www.googleapis.com/auth/ndev.clouddns.readwrite https://www.googleapis.com/auth/devstorage.full_control https://www.googleapis.com/auth/userinfo.email]
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Billing client for path https://cloudbilling.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Build client for path https://cloudbuild.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud BigQuery client for path https://bigquery.googleapis.com/bigquery/v2/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud CloudFunctions Client for path https://cloudfunctions.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud BigtableAdmin for path https://bigtableadmin.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Source Repo client for path https://sourcerepo.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Spanner client for path https://spanner.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Dataproc client for path https://dataproc.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Dataproc Beta client for path 
2020/10/06 23:48:50 [INFO] Instantiating Filestore client for path https://file.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud IoT Core client for path 
2020/10/06 23:48:50 [INFO] Instantiating App Engine client for path https://appengine.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Cloud Composer client for path https://composer.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Service Networking client for path https://servicenetworking.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Storage Transfer client for path https://storagetransfer.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud Healthcare client for path https://healthcare.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud CloudIdentity client for path https://cloudidentity.googleapis.com/
2020/10/06 23:48:50 [INFO] Instantiating Google Cloud KMS client for path https://cloudkms.googleapis.com/
2020/10/06 23:48:50 [DEBUG] Retry Transport: starting RoundTrip retry loop
2020/10/06 23:48:50 [DEBUG] Retry Transport: request attempt 0
2020/10/06 23:48:50 [DEBUG] Retry Transport: Stopping retries, last request failed with non-retryable error: empty non-retryable error received. This is a bug with the Terraform provider and should be reported as a GitHub issue in the provider repository.
2020/10/06 23:48:50 [DEBUG] Retry Transport: Returning after 1 attempts
2020/10/06 23:48:50 [DEBUG] Retry Transport: starting RoundTrip retry loop
2020/10/06 23:48:50 [DEBUG] Retry Transport: request attempt 0
2020/10/06 23:48:51 [DEBUG] Retry Transport: Stopping retries, last request failed with non-retryable error: empty non-retryable error received. This is a bug with the Terraform provider and should be reported as a GitHub issue in the provider repository.
2020/10/06 23:48:51 [DEBUG] Retry Transport: Returning after 1 attempts
--- PASS: TestAccDataSourceGoogleKmsCryptoKey_basic (25.81s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta 27.632s 
```

Reply:
One of the many reasons I'm proposing this change for v4 of the provider. https://github.com/hashicorp/terraform-provider-google/issues/7332

Reply:
[getTestCredsFromEnv()](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/utils/provider_test.go.erb#L883) is returning `"true"` because you have `GOOGLE_USE_DEFAULT_CREDENTIALS=true`. 

I believe we won't have to remove this line if we modify that function to return `""` instead when `multiEnvSearch(credsEnvVars)` returns `true`. Setting `GOOGLE_USE_DEFAULT_CREDENTIALS` should actually blank out the credentials field so that [config.go won't attempt to load it](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/third_party/terraform/utils/config.go.erb#L786).

Reply:
fixed

Reply:
Thanks @upodroid !